### ISSUE TITLE
Software architecture document

### ISSUE THREAD

**Issue by lamara11** [amosproj/amos2024ss01-xcelerator-demo-app#7]
## User story
1. As a developer
2. I need to prepare a description of the planned software architecture; The document should include:
  2.1. An overview diagram of runtime components (one page)
  2.2. An overview diagram of code components (one page)
  2.3. A summary of the underlying technology stack (one page)
  2.4. A textual explanation of the diagrams and choices (one page)
3. So that we have a definition of the components of the project

**Acceptance criteria**
- [x] All diagrams are available and reviewed by all developers. 
- [x] Underlying technology stack is agreed upon between developers. 
- [x] All diagrams and technology stack are explained. 
- [x] The architecture was presented to the Industry Partner
- [x] The software architecture document is uploaded to sprint-02/software-architecture.pdf

## Definition of done (DoD)

- [x] Added only after week 5
- [x] The same for all features
- [x] Here goes the project specific part

## DoD general criteria

- [x] Feature has been fully implemented
- [x] Feature has been merged into the mainline
- [x] All acceptance criteria were met
- [x] Product owner approved features
- [x] All tests are passing
- [x] Developers agreed to release

**Comment by sn0wcat** [comment]
@Sabo2k my suggestion is to use draw.io diagrams which can be checked in into the project and a markdown file (e.g. `architecture.md` with the description, which can then be converted to PDF or something similar if you guys need it as a deliverable for the university reviews.

**Comment by sn0wcat** [comment]
> **Acceptance criteria** All diagrams are available and reviewed by all developers. Underlying technology stack is agreed upon between developers. All diagrams and technology stack are explained.

+ The architecture was presented to the Industry Partner :)
